File: src/main/java/org/tindalos/principle/domain/detector/adp/ADPResult.java
package org.tindalos.principle.domain.detector.adp;

import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.Set;

import org.tindalos.principle.domain.core.Cycle;
import org.tindalos.principle.domain.core.PackageReference;
import org.tindalos.principle.domain.coredetector.CheckResult;
import org.tindalos.principle.domain.expectations.ADP;

import com.google.common.collect.Lists;
import com.google.common.collect.Maps;

public class ADPResult implements CheckResult {

    private final List<Cycle> cycles;
	private final ADP expectation;
    private final Map<PackageReference, Set<Cycle>> cyclesByBreakingPoints;

    public ADPResult(List<Cycle> cycles, Map<PackageReference, Set<Cycle>> cyclesByBreakingPoints, ADP adpExpectation) {
        this.cyclesByBreakingPoints = Maps.newHashMap(cyclesByBreakingPoints) ;
        this.expectation = adpExpectation;
		this.cycles = Lists.newArrayList(cycles);
		Collections.sort(this.cycles);
    }
    
    public List<Cycle> getCycles() {
        return cycles;
    }

	public boolean expectationsFailed() {
		return cyclesByBreakingPoints.size() > expectation.getViolationsThreshold();
	}

    public Integer getThreshold() {
        return expectation.getViolationsThreshold();
    }
    
    public Map<PackageReference, Set<Cycle>> getCyclesByBreakingPoints() {
        return cyclesByBreakingPoints;
    }

}

File: src/main/java/org/tindalos/principle/domain/detector/adp/ADPResult.scala
package org.tindalos.principle.domain.detector.adp

import org.tindalos.principle.domain.core.{Cycle, PackageReference}
import org.tindalos.principle.domain.coredetector.AnalysisResult
import org.tindalos.principle.domain.detector.structure.Structure.NodeId
import org.tindalos.principle.domain.expectations.ADP


class ADPResult(val cyclesByBreakingPoints: Map[PackageReference, Set[Cycle]], val expectation: ADP) extends AnalysisResult {

  val threshold = expectation.violationsThreshold

  override def expectationsFailed() = cyclesByBreakingPoints.size > threshold

}
